include/networkit/randomization/GlobalCurveball.hpp

```diff
@@ -27,14 +27,21 @@ class GlobalCurveball : public Algorithm {
      * @param G                        Undirected and unweighted graph to be randomized
      * @param number_of_global_trades  Number of global trades to be executed (each edge
      *                                 is considered exactly twice per global traded)
+     * @param allowSelfLoops           May only be set if graph is directed.
      * @param degreePreservingShufflePreprocessing Execute DegreePreservingShuffle
      *                                 (see Algorithm for description) as a preprocessing
      *                                 step. This is more efficient than calling the algorithm
      *                                 explicitly.
+     *
+     * @note Self loops can only be realized for directed graphs.
```

Comment:
Is this a limitation of Curveball?

Reply:
Yes, AFAIK neither Curveball nor EdgeSwitching give uniform samples for non-simple undirected graphs.